pkg/builder/controller.go

```diff
@@ -344,12 +340,13 @@ func (blder *Builder) doController(r reconcile.Reconciler) error {
 	globalOpts := blder.mgr.GetControllerOptions()
 
 	ctrlOptions := blder.ctrlOptions
-	if ctrlOptions.Reconciler == nil {
-		ctrlOptions.Reconciler = r
+	if ctrlOptions.Reconciler != nil {
+		return errors.New("reconciler was already set via WithOptions()")
```

Comment:
I think it makes sense to return an error when two reconcilers are set via the buidler and not just silently overwrite